.evergreen/config_generator/components/cse/darwinssl.py

```diff
@@ -0,0 +1,75 @@
+from shrub.v3.evg_build_variant import BuildVariant
+from shrub.v3.evg_task import EvgTaskRef
+
+from config_generator.etc.compile import generate_compile_tasks
+
+from config_generator.etc.cse.compile import CompileCommon
+from config_generator.etc.cse.test import generate_test_tasks
+
+
+SSL = 'darwinssl'
+TAG = f'cse-matrix-{SSL}'
+
+
+# pylint: disable=line-too-long
+# fmt: off
+COMPILE_MATRIX = [
+    ('macos-1014', 'clang', None, ['cyrus']),
+]
+
+# TODO (CDRIVER-3789): test cse with the 'sharded' topology.
+TEST_MATRIX = [
+    ('macos-1014', 'clang', None, 'cyrus', ['auth'], ['server', 'replica' ], ['4.2', '4.4', '5.0']),
+
+    # Test 6.0+ with a replica set since Queryable Encryption does not support the 'server' topology.
+    ('macos-1014', 'clang', None, 'cyrus', ['auth'], ['server', 'replica' ], ['6.0', 'latest']),
```

Comment:
The replica + 6.0 test is [currently failing](https://spruce.mongodb.com/task/mongo_c_driver_cse_matrix_darwinssl_cse_sasl_cyrus_darwinssl_macos_1014_clang_test_6.0_replica_auth_patch_8b5a14e5a35fb7b365a1a53f24c933f8c497f337_63e6a4b132f4174313baa9be_23_02_10_20_10_28/logs?execution=0) due to the [following error](https://parsley.mongodb.com/evergreen/mongo_c_driver_cse_matrix_darwinssl_cse_sasl_cyrus_darwinssl_macos_1014_clang_test_6.0_replica_auth_patch_8b5a14e5a35fb7b365a1a53f24c933f8c497f337_63e6a4b132f4174313baa9be_23_02_10_20_10_28/0/task?bookmarks=0,806&shareLine=765) during `/client_side_encryption/legacy/fle2-EncryptedFields-vs-jsonSchema`:

```
test error in: ./mongoc/src/libmongoc/tests/json-test-operations.c 691:check_result()
Error in "encryptedFields is preferred over jsonSchema" test : expected 1 keys, not 0
Expected:
{ "0" : { "_id" : 1, "encryptedIndexed" : "123" } }
Actual:
{ }
```

The replica + latest test is [also failing](https://spruce.mongodb.com/task/mongo_c_driver_cse_matrix_darwinssl_cse_sasl_cyrus_darwinssl_macos_1014_clang_test_latest_replica_auth_patch_8b5a14e5a35fb7b365a1a53f24c933f8c497f337_63e6a4b132f4174313baa9be_23_02_10_20_10_28/logs?execution=0) due to the [following error](https://parsley.mongodb.com/evergreen/mongo_c_driver_cse_matrix_darwinssl_cse_sasl_cyrus_darwinssl_macos_1014_clang_test_latest_replica_auth_patch_8b5a14e5a35fb7b365a1a53f24c933f8c497f337_63e6a4b132f4174313baa9be_23_02_10_20_10_28/0/task?bookmarks=0,730&shareLine=689) during `/client_side_encryption/legacy/fle2-Range-DecimalPrecision-FindOneAndUpdate`:

```
error initializing cipher: Unimplemented (-4305). CTR mode is only supported on macOS 10.15+
```

It seems like this issue is exclusive to either DarwinSSL or the macos-1014 distro, as [no other task](https://spruce.mongodb.com/version/63e6a4b132f4174313baa9be/tasks?page=0&sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC&taskName=cse-.%2A-test-%286%5C.0%7Clatest%29-replica-auth) with the same test parameters are failing.

We could attempt to address these issues in this PR, or defer them to followup work (new JIRA tickets).

Reply:
> The replica + 6.0 test is [currently failing](https://spruce.mongodb.com/task/mongo_c_driver_cse_matrix_darwinssl_cse_sasl_cyrus_darwinssl_macos_1014_clang_test_6.0_replica_auth_patch_8b5a14e5a35fb7b365a1a53f24c933f8c497f337_63e6a4b132f4174313baa9be_23_02_10_20_10_28/logs?execution=0) due to the [following error](https://parsley.mongodb.com/evergreen/mongo_c_driver_cse_matrix_darwinssl_cse_sasl_cyrus_darwinssl_macos_1014_clang_test_6.0_replica_auth_patch_8b5a14e5a35fb7b365a1a53f24c933f8c497f337_63e6a4b132f4174313baa9be_23_02_10_20_10_28/0/task?bookmarks=0,806&shareLine=765) during `/client_side_encryption/legacy/fle2-EncryptedFields-vs-jsonSchema`:
> 
> ```
> test error in: ./mongoc/src/libmongoc/tests/json-test-operations.c 691:check_result()
> Error in "encryptedFields is preferred over jsonSchema" test : expected 1 keys, not 0
> Expected:
> { "0" : { "_id" : 1, "encryptedIndexed" : "123" } }
> Actual:
> { }
> ```

Additional logging [shows the "insertOne" operation fails with this message](https://spruce.mongodb.com/task/mongo_c_driver_cse_matrix_darwinssl_cse_sasl_cyrus_darwinssl_macos_1014_clang_test_6.0_replica_auth_patch_8b5a14e5a35fb7b365a1a53f24c933f8c497f337_63ea7a2c3066153b5a6afa86_23_02_13_17_58_06/logs?execution=0):

> CTR mode is only supported on macOS 10.15+

The test runner ignores the error from the "insertOne" operation. The "insertOne" operation [does not include a result assertion](https://github.com/mongodb/specifications/blob/31bbf60c4f3008d77fc5ca5287b165e5ea97a4a6/source/client-side-encryption/tests/legacy/fle2-EncryptedFields-vs-jsonSchema.yml#L21-L23). Operations without result assertions are [not checked for errors in the test runner](https://github.com/mongodb/mongo-c-driver/blob/6a7692a8357df279dfbfd9c89ed8724ec0aacd9f/src/libmongoc/tests/json-test-operations.c#L633-L637).

I suggest running the `cse` tests with macOS 10.15. MONGOCRYPT-385 introduced the Queryable Encryption feature that is unavailable on macOS 10.14.

Continue running the non-`cse` tests with macOS 10.14, since it is listed under [Supported Platforms](https://mongoc.org/libmongoc/current/installing.html).